### PR TITLE
MAINT: Remove _util._valarray

### PR DESCRIPTION
**Pull request by peterbell10**
#### What does this implement/fix?
As far as I can tell, `_util._valarray` does the exact same thing as `numpy.full` but in a less efficient way.

```python
In [3]: %timeit _valarray((100, 100), 10, 'd')                                                                                
18.2 µs ± 174 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)                                                                                         

In [4]: %timeit np.full((100, 100), 10, 'd')                                                                                  
5.79 µs ± 75.4 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

### CODE REVIEW

File: scipy/_lib/_util.py [hunk 1]
```diff
@@ -55,7 +43,7 @@ def _lazywhere(cond, arrays, f, fillvalue=None, f2=None):
     arrays = np.broadcast_arrays(*arrays)
     temp = tuple(np.extract(cond, arr) for arr in arrays)
     tcode = np.mintypecode([a.dtype.char for a in arrays])
-    out = _valarray(np.shape(arrays[0]), value=fillvalue, typecode=tcode)
```
Comment (sethtroisi) [inline scipy/_lib/_util.py:58]: In my trivial example this seems to speedup `_lazywhere` by ~ 13%!

```
In [1]: import scipy.special as sc
In [2]: from scipy._lib._util import _lazywhere
In [3]: import numpy as np
In [4]: t = 10 ** np.arange(10)
In [5]: %timeit _lazywhere((t > 0), (t,), lambda v: sc.log1p(v), -np.inf)
46.9 µs ± 1.23 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [6]: %timeit sc.log1p(t)
1.68 µs ± 55.3 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

after

```
In [5]: %timeit _lazywhere((t > 0), (t,), lambda v: sc.log1p(v), -np.inf)
40.8 µs ± 522 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [6]: %timeit sc.log1p(t)
1.88 µs ± 35.9 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

File: scipy/_lib/_util.py [hunk 2]
```diff
@@ -16,18 +16,6 @@ class Generator():  # type: ignore[no-redef]
         pass
 
 
-def _valarray(shape, value=np.nan, typecode=None):
```
Comment (sethtroisi) [inline scipy/_lib/_util.py:19]: This is double underscored `_util` and `_valarray` but should we possible alias it to `np.full` for one release with a runtime warning?

File: scipy/stats/_distn_infrastructure.py [hunk 3]
```diff
@@ -1052,7 +1051,7 @@ def stats(self, *args, **kwds):
         args = tuple(map(asarray, args))
         cond = self._argcheck(*args) & (scale > 0) & (loc == loc)
         output = []
-        default = valarray(shape(cond), self.badvalue)
+        default = np.full(shape(cond), self.badvalue)
```
Comment (sethtroisi) [inline scipy/stats/_distn_infrastructure.py:1054]: nit: consider adding add `fill_value=` to be consistent with other uses in this file

**Review by sethtroisi — APPROVED** [review]

**Review by rgommers — APPROVED** [review]
Nice catch, LGTM too.

**Pull request merged**